tuna.go

```diff
@@ -415,8 +415,9 @@ func (c *Common) startUDPReaderWriter(conn *EncryptUDPConn, toAddr *net.UDPAddr,
 			n, from, encrypted, err = conn.ReadFromUDPEncrypted(buffer)
 			if err != nil {
 				log.Println("Couldn't receive data:", err)
-				if strings.Contains(err.Error(), "use of closed network connection") {
-					c.udpCloseChan <- struct{}{}
+				if errors.Is(err, io.ErrClosedPipe) {
```

Comment:
`io.ErrClosedPipe` is `io: read/write on closed pipe`, different from `use of closed network connection`

Reply:
> ErrClosedPipe

In this case, `io.ErrClosedPipe` is the error when read from a closed UDP connection